File: c/cz/mg/vulkan/PFN_vkAllocateDescriptorSets.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkAllocateDescriptorSets_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pAllocateInfo, jlong pDescriptorSets, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkAllocateDescriptorSets f = (PFN_vkAllocateDescriptorSets)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkDescriptorSetAllocateInfo*)jniLongToPointer(pAllocateInfo)),
        ((VkDescriptorSet*)jniLongToPointer(pDescriptorSets))
    );
}

File: cz/mg/vulkan/PFNvkAllocateDescriptorSets.java
package cz.mg.vulkan;

public class PFNvkAllocateDescriptorSets extends VkFunctionPointer {
    public PFNvkAllocateDescriptorSets() {
    }

    public PFNvkAllocateDescriptorSets(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkAllocateDescriptorSets(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkAllocateDescriptorSets(long value) {
        setValue(value);
    }

    public PFNvkAllocateDescriptorSets(VkInstance instance) {
        super(instance, new VkString("vkAllocateDescriptorSets"));
    }

    public void call(VkDevice device, VkDescriptorSetAllocateInfo pAllocateInfo, VkDescriptorSet pDescriptorSets, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getSinkAddress(), pAllocateInfo != null ? pAllocateInfo.getVkAddress() : VkPointer.NULL, pDescriptorSets != null ? pDescriptorSets.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddress());
    }

    private static native void call(long vkaddress, long device, long pAllocateInfo, long pDescriptorSets, long rval);
}
